gsa/src/web/pages/scanners/dialog.js

```diff
@@ -182,8 +186,16 @@ class ScannerDialog extends React.Component {
     const show_cred_info =
       isDefined(scanner) &&
       isDefined(scanner.credential) &&
-      scanner.credential.type === CLIENT_CERTIFICATE_CREDENTIAL_TYPE;
-    const isGmpScannerType = type === GMP_SCANNER_TYPE;
+      scanner.credential.credential_type === CLIENT_CERTIFICATE_CREDENTIAL_TYPE;
+
+    const isOspSensorType = type === OSP_SENSOR_SCANNER_TYPE;
+    const isOspScannerType = type === OSP_SCANNER_TYPE;
+
+    let {credential_id} = this.props;
+
+    if (isOspSensorType) {
+      credential_id = '';
```

Comment:
@saberlynx could you check if undefined fits now with Timos patch https://github.com/greenbone/gvmd/pull/759 too? Personally I don't want to send special values like '0' or '' for "no value". Instead the API should just get not value.

Reply:
Do you mean we should be able to do something like credential_id = undefined or credential_id = null? I was able to make it work with '0', but not '', null or undefined.

Reply:
Nevermind. I misunderstood. I got rid of line 196-198 entirely and it works great.

Reply:
Test results:
credential_id = null => Given credential_id was invalid
crednetial_id = '' => Given credential_id was invalid
credential_id = undefined => Given credential_id was invalid
credential_id = '0' => successfully saved without a credential

getting rid of lines 196-198 entirely => scanner will save with default 'cc' credential.

Reply:
Ah ok I guess now gsad's validation must be adjusted to allow an empty credential_id for this scanner type.